vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpacesCustomOrderManager.java

```diff
@@ -0,0 +1,145 @@
+                SpacesCustomOrder spaceCustomOrder = spaceCustomOrderOptional.get();
+                spaceCustomOrder.setCustomOrder(customOrder);
+                spacesCustomOrderNewList.add(spaceCustomOrder);
+            }
+        }
+        spacesCustomOrderRepository.saveAll(spacesCustomOrderNewList);
+    }
+    
+    /**
+     * This method persists all the published spaces into the SpacesCustomOrder table 
+     * This method is called every time we go to custom order tab as that will 
+     * ensure the table is populated with latest published spaces before changing the order
+     */    
+    @Override
+    public void persistPublishedSpacesToSpacesCustomOrder() {
+        List<ISpace> spaces = (List<ISpace>) spaceManager.getSpacesWithStatus(SpaceStatus.PUBLISHED);
+        spaces.addAll(spaceManager.getSpacesWithStatus(null));
+        List<SpacesCustomOrder> spacesCustomOrderNewList = new ArrayList<>();
+        int customOrder = 0;
+        if(spacesCustomOrderRepository.findMaxCustomOrder() != null) {
+            customOrder = spacesCustomOrderRepository.findMaxCustomOrder();
+        }
+        for(ISpace space : spaces) {
+            Optional<SpacesCustomOrder> spaceCustomOrderOptional = spacesCustomOrderRepository.findBySpace_Id(space.getId());
+            if(!spaceCustomOrderOptional.isPresent()) {
+                spacesCustomOrderNewList.add(new SpacesCustomOrder(space, ++customOrder));
+                logger.info("custom order is {}", customOrder);
+            }
+        }
+        spacesCustomOrderRepository.saveAll(spacesCustomOrderNewList);
+    }
+    
+    @Override
+    public List<SpacesCustomOrder> findAll(){
+        return (List<SpacesCustomOrder>) spacesCustomOrderRepository.findAll();
+    }
+    
+    @Override
+    public void updateStatusChange(ISpace space, SpaceStatus status) {
+        if(status == SpaceStatus.PUBLISHED || status == null) {
+            updateStatusChangeToPublished(space);
+        }else {
+            updateStatusChangeToUnpublished(space);
+        }
+    }
+   
+    /**
+     * This method updates the custom order of newly published space to the max order
+     * @param space
+     */
+    @Override
+    public void updateStatusChangeToPublished(ISpace space) {
+        Optional<SpacesCustomOrder> spaceCustomOrderOptional = spacesCustomOrderRepository.findBySpace_Id(space.getId());
+        if(!spaceCustomOrderOptional.isPresent()) {
+            int order = spacesCustomOrderRepository.findMaxCustomOrder() + 1;
+            SpacesCustomOrder spaceCustomOrder = new SpacesCustomOrder(space, order);
+            spaceCustomOrder.setCustomOrder(order);
+            spacesCustomOrderRepository.save(spaceCustomOrder);
+        }
+        
+    }
+    
+    /**
+     * This method updates the custom order of existing spaces when a space is changed to unpublished
+     * @param space
+     */
+    @Override
+    public void updateStatusChangeToUnpublished(ISpace space) {
```

Comment:
I don't think this is necessary; as long as when presented it is checked that a space is published, they can be part of the ordering even when they are unpublished.